### ISSUE TITLE
[FYI] loader.js cannot find monaco-editor when packaging for nsis

### ISSUE THREAD

**Issue by buddythumbs** [superRaytin/react-monaco-editor#83]
Hi,

I was packaging all along with windows target set to portable:
In my package.json (top level):

```
"build: {
    "win":{
        "target":"portable"
    }
}
```

However, after finding I needed nsis for a more professional experience I changed to:

```
"build: {
    "win":{
        "target":"nsis"
    }
}
```

and the editor stopped working.

The issue was that the loader.js script was looking in `C/Users/username/Appdata/Local/Programs/AppID/resources/node_modules/monaco-editor/min/vs` and that folder was not being created in the install using the nsis packaging method.

The way I got around this was to change the `webpack.base.config .js` to have the following output for vs:
```
    new CopyWebpackPlugin([
      {
        from: 'node_modules/monaco-editor/min/vs',
        to: 'resources/node_modules/monaco-editor/vs',
      }
    ]),
```

with the resources folder in root of project.

Then in the `package.json` I needed to include these extra resources when building with electron builder. This was achieved with the following code:

```
  "build": {
    "productName": "OpenSO",
    "asar": false,
    "appId": "asml.ireland.OpenSO",
    "files": [
      "dist/**/*",
      "node_modules/**/*",
      "app.html",
      "main.prod.js",
      "main.prod.js.map",
      "package.json",
      "assests/**/*",
      "vbs/**/*",
      "app/data/**/*",
      "resources/node_modules/**/*"
    ],
    "dmg": {
      "contents": [
        {
          "x": 130,
          "y": 220
        },
        {
          "x": 410,
          "y": 220,
          "type": "link",
          "path": "/Applications"
        }
      ]
    },
    "linux": {
      "target": [
        "deb",
        "AppImage"
      ]
    },
    "win": {
      "target": "nsis"
    },
    "directories": {
      "buildResources": "resources",
      "output": "release"
    },
    "extraResources": [
      "app/assets/**/*",
      "app/vbs/**/*",
      "vs/**/*",
      "app/data/**/*",
      "node_modules/monaco-editor/**/*"
    ]
```
All these files mentioned here in the `extraResources` get added to that resources folder during install so the file now exists!

I am just putting this here because I am sure someone else will encounter a similar problem.

Also if anyone knows a more elegant way of doing this (with webpack maybe) please comment as I am hacking my way through getting this to work :)

**Comment by nickaversano** [comment]
I was able to get this to work by doing the following:

In my `webpack.config.js`, adding the CopyWebpackPlugin as [described here](https://github.com/superRaytin/react-monaco-editor#using-with-webpack).

Then in my `CodeEditor.js` file I added the following `requireConfig`:

```javascript
import { remote } from 'electron'
const appPath = remote.app.getAppPath()

export default class CodeEditor extends Component {
  render() {
    return (
        <MonacoEditor
          requireConfig={{
            url: `${appPath}/renderer/vs/loader.js`,
            baseUrl: `${appPath}/renderer`
          }}
        />
    )
  }
}
```

Note that I'm using the two `package.json` setup and my webpack config is set up to build my `src` files into the `app/renderer` path (where app is the folder containing my Electron app).

Hope that helps! Was really banging my head trying to figure this out
